ha/cli/exec/clusterExecutor.py

```diff
@@ -129,50 +136,156 @@ def execute(self):
         raise HAUnimplemented("This operation is not implemented.")
 
-    def execute(self) -> str:
+    def execute(self) -> None:
         raise HAUnimplemented("This operation is not implemented.")
 
 class ClusterActiveExecutor(CommandExecutor):
-    def validate(self) -> str:
+    def validate(self) -> bool:
         raise HAUnimplemented("This operation is not implemented.")
 
-    def execute(self) -> str:
+    def execute(self) -> None:
         raise HAUnimplemented("This operation is not implemented.")
 
 class ClusterListExecutor(CommandExecutor):
-    def validate(self) -> str:
+    def validate(self) -> bool:
         raise HAUnimplemented("This operation is not implemented.")
 
-    def execute(self) -> str:
+    def execute(self) -> None:
         raise HAUnimplemented("This operation is not implemented.")
 
 class ClusterStatusExecutor(CommandExecutor):
-    def validate(self) -> str:
+    def validate(self) -> bool:
         raise HAUnimplemented("This operation is not implemented.")
 
-    def execute(self) -> str:
+    def execute(self) -> None:
         raise HAUnimplemented("This operation is not implemented.")
 
-class ClusterAddExecutor(CommandExecutor):
-    def validate(self) -> str:
-        raise HAUnimplemented("This operation is not implemented.")
+class ClusterNodeAddExecutor(CommandExecutor):
+    '''
+        Module which will accept the request for cluster add node
+        functinality and which is responsible for delegating that request
+        to Cluster Manager
+    '''
 
-    def execute(self) -> str:
-        raise HAUnimplemented("This operation is not implemented.")
+    def __init__(self):
+        '''Init Method'''
+        super(ClusterNodeAddExecutor, self).__init__()
+        self._pcs_cluster_controller = PcsClusterController()
+        self._op = Output()
+        self._args = None
+
+    def parse_cluster_args(self) -> bool:
+        '''
+           Parses the command line args.
+           Return: argparse
+        '''
+        parser = argparse.ArgumentParser(prog='cluster add node')
+        parser.add_argument("cluster", help="Module")
+        parser.add_argument("add", help="action to be performed")
+        parser.add_argument("node", help="component on which action to be performed")
+        group = parser.add_mutually_exclusive_group(required='True')
+        group.add_argument('--nodeid', action='store', \
+                            help='ID of a node which needs to be added in a cluster')
+        group.add_argument('--descfile', action='store', \
+                            help='A file which describes the node to be added in a cluster', \
+                            type=self._is_file_extension_valid)
+        parser.add_argument('--username', help='cluster username', required=True)
+        parser.add_argument('--password', help='cluster password', required=True)
+        parser.add_argument('--json', help='Required output fomat', action='store_true')
```

Comment:
*format

Reply:
Done. Thanks for checking this and informing.